acrg/BC/BC_CAMS.py

```diff
@@ -341,46 +392,61 @@ def create_CAMS_BC(ds, fp_lat, fp_lon, fp_height, date, domain, species="ch4",
-                         fp_lon, species, lonorlat='longitude', verbose=False).rename({species : 'vmr_n'})   
-    vmr_s = interplonlat(interpheight(south, fp_height, species, lonorlat='longitude'),
-                         fp_lon, species, lonorlat='longitude', verbose=False).rename({species : 'vmr_s'}) 
-    vmr_e = interplonlat(interpheight(east, fp_height, species, lonorlat='latitude'),
-                         fp_lat, species, lonorlat='latitude', verbose=False).rename({species : 'vmr_e'}) 
-    vmr_w = interplonlat(interpheight(west, fp_height, species, lonorlat='latitude'),
-                         fp_lat, species, lonorlat='latitude', verbose=False).rename({species : 'vmr_w'})      
-
-    write_CAMS_BC_tonetcdf(vmr_n, vmr_e, vmr_s, vmr_w, date, species, domain, outdir, gridsize, from_climatology=from_climatology) 
+    north = ds.sel(latitude  = ds.coords[lat_var][lat_n],
+                   longitude = slice(ds.coords[lon_var][lon_w],ds.coords[lon_var][lon_e])).drop_vars([lat_var])
+    south = ds.sel(latitude  = ds.coords[lat_var][lat_s],
+                   longitude = slice(ds.coords[lon_var][lon_w],ds.coords[lon_var][lon_e])).drop_vars([lat_var])
+    east  = ds.sel(longitude = ds.coords[lon_var][lon_e],
+                   latitude  = slice(ds.coords[lat_var][lat_s],ds.coords[lat_var][lat_n])).drop_vars([lon_var])
+    west  = ds.sel(longitude = ds.coords[lon_var][lon_w],
+                   latitude  = slice(ds.coords[lat_var][lat_s],ds.coords[lat_var][lat_n])).drop_vars([lon_var])
+    
+    vmr_n = interplonlat(interpheight(north, fp_height, species=species, lonorlat=lon_var),
+                         fp_lon, species=species, lonorlat=lon_var, height_var=height_var, verbose=False).rename({species : 'vmr_n'})   
+    vmr_s = interplonlat(interpheight(south, fp_height, species=species, lonorlat=lon_var),
+                         fp_lon, species=species, lonorlat=lon_var, height_var=height_var, verbose=False).rename({species : 'vmr_s'}) 
+    vmr_e = interplonlat(interpheight(east, fp_height, species=species, lonorlat=lat_var),
+                         fp_lat, species=species, lonorlat=lat_var, height_var=height_var, verbose=False).rename({species : 'vmr_e'}) 
+    vmr_w = interplonlat(interpheight(west, fp_height, species=species, lonorlat=lat_var),
+                         fp_lat, species=species, lonorlat=lat_var, height_var=height_var, verbose=False).rename({species : 'vmr_w'})      
+
+    write_CAMS_BC_tonetcdf(vmr_n = vmr_n,
+                           vmr_e = vmr_e,
+                           vmr_s = vmr_s,
+                           vmr_w = vmr_w,
+                           st_date = date,
+                           domain = domain,
+                           outdir = outdir,
+                           gridsize = gridsize,
+                           species = species,
+                           from_climatology = from_climatology) 
 
 def makeCAMSBC(domain, start, end,
```

Comment:
Same comment as above for many input variables (including `cams_directory` - fine to set this to the global default which can be defined above).